umm-spec-lib/src/cmr/umm_spec/xml_to_umm_mappings/dif10.clj

```diff
@@ -197,6 +197,30 @@
         :OnlineResource (when-let [linkage (value-of data-set-citation "Online_Resource")]
                           {:Linkage linkage})}))))
 
+(defn- parse-archive-dist-info
+  "Parses ArchiveAndDistributionInformation out of DIF XML into UMM-C"
+  [doc]
+  (let [distributions
+        (for [distribution (select doc "/DIF/Distribution")
```

Comment:
Nitpick: move this `for` into its own function to keep logic out of `let` blocks